lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -566,14 +605,18 @@ def generate_clone_spec(src_config)
 
     if config[:customization_ips]
       if get_config(:customization_gw)
-        cust_spec.nicSettingMap = config[:customization_ips].split(',').map { |i| generate_adapter_map(i, get_config(:customization_gw)) }
+        cust_spec.nicSettingMap = config[:customization_ips].split(',').map.with_index { |i, index|
+          generate_adapter_map(i, get_config(:customization_gw), mac_list[index])
+        }
       else
-        cust_spec.nicSettingMap = config[:customization_ips].split(',').map { |i| generate_adapter_map(i) }
+        cust_spec.nicSettingMap = config[:customization_ips].split(',').map.with_index { |i, index|
```

Comment:
Same comment as above. Actually, both these clauses look like they could be collapsed, if `:customization_gw` isn't specified then `get_config` will return `nil`, so we're back to `generate_adapter_map(i, get_config(:customization_gw), mac_list[index])` in either case.